### PR TITLE
Don't throw an exception if none LS was found

### PR DESCRIPTION
**Pull request by svor**
Signed-off-by: Valeriy Svydenko <vsvydenk@redhat.com>

### What does this PR do?
1. Add a way to check if jdt.ls installer exists in workspace's configuration;
2. Add initialization methods which is called when workspace is ready;
3. Don't ask for Java compiler options when workspace is running.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/11091

### CODE REVIEW

**Review by tolusha — APPROVED** [review]

**Review by tsmaeder — CHANGES_REQUESTED** [review]
Generally, I think we should do most of the work in the back end.

**Comment by svor** [comment]
@tsmaeder I've updated the PR. Please take another look at it.

File: plugins/plugin-java/che-plugin-java-server/src/main/java/org/eclipse/che/plugin/java/languageserver/JavaLanguageServerExtensionService.java [hunk 1]
```diff
@@ -955,7 +955,9 @@ public OrganizeImportsResult organizeImports(OrganizeImportParams organizeImport
    */
   public JavaCoreOptions getJavaCoreOptions(List<String> filter) {
     Type type = new TypeToken<JavaCoreOptions>() {}.getType();
-    return doGetOne(Commands.GET_JAVA_CORE_OPTIONS_СOMMAND, new ArrayList<>(filter), type);
+    JavaCoreOptions result =
+        doGetOne(Commands.GET_JAVA_CORE_OPTIONS_СOMMAND, new ArrayList<>(filter), type);
+    return result == null ? new JavaCoreOptions() : result;
```
Comment (tsmaeder) [inline plugins/plugin-java/che-plugin-java-server/src/main/java/org/eclipse/che/plugin/java/languageserver/JavaLanguageServerExtensionService.java:960]: Why do we need the null check? Shoudn't this return a result or throw an exception?

Comment (svor) [reply plugins/plugin-java/che-plugin-java-server/src/main/java/org/eclipse/che/plugin/java/languageserver/JavaLanguageServerExtensionService.java:960]: It's better to return empty map of java compiler options than null, if jdt.ls wasn't initialized

**Review by tsmaeder — APPROVED** [review]

**Pull request merged**